### ISSUE TITLE
Feature request: initialize super class with super call

### ISSUE THREAD

**Issue by NeilGirdhar** [charliermarsh/ruff#3403]
A common error with Python inheritance is to try to explicitly call base class methods:
```python
from typing import Any

class C:
    def __init__(self, x: int):
        self.x = x


class D(C):
    def __init__(self) -> None:
        C.__init__(self, 1)  # Bad!
        # super().__init__(x=1)  # Good.


class E(C):
    def __init__(self, **kwargs: Any) -> None:
        super().__init__(**kwargs)
        self.y = 2


class F(D, E):
    pass


f = F()
print(f.x, f.y)  # AttributeError: 'F' object has no attribute 'y'
```
The problem with `D`'s definition is that `E` and `F` can come along and now `E`'s initializer is not called.

This is a problem with all methods (not just `__init__`) although `__init__` is the most common place this fails.

In general, I think it should be a Ruff error to call a base class method explicitly rather than through `super`.  Example of [this error](https://github.com/pandas-dev/pandas/blob/a07cb65459f24446e82354854ff6658f29414c0a/pandas/core/frame.py#L852).

The same problem would manifest with `__enter__`, `__exit__`, or any user-defined method that exhibits cooperative inheritance.

**Comment by charliermarsh** [comment]
We have UP008, which handles the _arguments_ portion to `__super__`, but I guess this is different in that we don't catch the `C` in `C.__init__`. Is that right?

**Comment by NeilGirdhar** [comment]
The problem is that you should rarely (if ever) call your superclass by its name: `C.__init__(...`.  You should instead call it like this `super().__init__(...`

**Comment by Avasam** [comment]
I've been upgrading pywin32's legacy code over the past year, and this is also something I noticed. Would love an autofix (even if marked unsafe) for the hundreds of violations ^^

![Image](https://github.com/user-attachments/assets/e0ae8686-06b7-43c3-8f68-a42361dd035f)

**Comment by NeilGirdhar** [comment]
@Avasam Wow, that looks painful.  Too bad Ruff hadn't warned that developer before they did that!

**Comment by Avasam** [comment]
> Too bad Ruff hadn't warned that developer before they did that!

What do you mean? Ruff hasn't done anything. I'm +1-ing the feature request to catch this (to prevent regressions), and asking for an autofix at the same time.

**Comment by NeilGirdhar** [comment]
@Avasam Yes, I understand (Thank you).   I just mean too bad Ruff didn't have this feature before someone made that mess for you to clean up 😄
